### PR TITLE
Integrate version compare API info into footer_html endpoint

### LINKED ISSUE #1463: Add Version Compare information to the RTD Footer Response
We want to unify the API calls for `/api/v1/<project>/highest/<version>` and `/api/v2/footer_html` in order to reduce the number of API calls.

### PR DESCRIPTION
**Pull request by gregmuellegger**
This integrates the data provided by `/api/v1/version/<project>/highest/<version>` into the `/api/v2/footer_html/` response. We can use this to avoid one API request per doc page view. The API v1 version endpoint is still in place for backwards compatibility.

I also refactored the doc-embed JavaScript into multiple files for easier maintainability.

Fixes #1463.

### CODE REVIEW

**Comment by ericholscher** [comment]
Big +1 on making our JS more managable.

File: readthedocs/core/static-src/core/js/doc-embed/footer.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            data.promo_data.image
+        )
+        if (promo) {
+            promo.display();
+        }
+    }
+}
+
+
+function setupBookmarkCSRFToken() {
```
Comment (ericholscher) [inline readthedocs/core/static-src/core/js/doc-embed/footer.js]: We can probably kill this for now -- we have killed the Bookmarking feature (it was half done, and probably not the most reasonable from a product perspective). However, this code does do some interesting things that we'll likely need to do when we actually look seriously at doing CNAME-supported javascript clients.

**Comment by ericholscher** [comment]
Looks like a solid refactor. :+1: -- Do we need to include any changes in the gulp configs or anything along those lines, or will it automagically just work?

**Comment by ericholscher** [comment]
:fireworks:

**Comment by ericholscher** [comment]
This broke when I tested it, so I've reverted it for now.

**Comment by gregmuellegger** [comment]
Sorry mate! :crying_cat_face: I have addressed the raised issues in #1507

**Pull request merged**